### PR TITLE
Add FalDbt class and use it where possible for magic functions

### PR DESCRIPTION
**Pull request by chamini2**
Initial ideas here. It is in working condition but can be refactored.
Tests should be fixed.

### CODE REVIEW

**Review by mkam039 — COMMENTED** [review]
It's very nice! 👏 👏 👏 

I got to test it from within a notebook and it worked without any problems.

Since we're losing the context, we need to somehow get information on available models and their states.

File: src/fal/cli.py [hunk 1]
```diff
@@ -3,6 +3,7 @@
 
 from dbt.logger import log_manager
 from dbt.config.profile import DEFAULT_PROFILES_DIR
+from fal.fal import FalDbt
```
[the remark this answers is not in the document]
Comment (chamini2) [reply src/fal/cli.py:6]: You can do `from fal import FalDbt` in the external script / notebook

**Comment by turbo1912** [comment]
> Since we're losing the context, we need to somehow get information on available models and their states.

Are we losing context? Users still have to tell us where their project is and we are able to gather all the information we need from the project.. 
```
profiles_dir="/Users/matteo/.dbt",
project_dir="/Users/matteo/Projects/fal/experiments/iris
```
but maybe when you say context, you are referring to something else?

**Comment by mkam039** [comment]
> > Since we're losing the context, we need to somehow get information on available models and their states.
> 
> Are we losing context? Users still have to tell us where their project is and we are able to gather all the information we need from the project..
> 
> ```
> profiles_dir="/Users/matteo/.dbt",
> project_dir="/Users/matteo/Projects/fal/experiments/iris
> ```
> 
> but maybe when you say context, you are referring to something else?

I meant the `context` variable.

**Pull request closed without merging**